src/main/java/org/broadinstitute/consent/http/service/MatchService.java

```diff
@@ -32,35 +24,136 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class DatabaseMatchingServiceAPI extends AbstractMatchingServiceAPI {
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
+public class MatchService {
+    private final MatchDAO matchDAO;
     private final ConsentDAO consentDAO;
+    private DataAccessRequestAPI accessAPI;
     private final ElectionDAO electionDAO;
-    private final DataAccessRequestDAO dataAccessRequestDAO;
-    private DatasetService datasetService;
-    private final WebTarget matchServiceTarget;
-    private final GenericType<ResponseMatchingObject> rmo = new GenericType<>(){};
     private final UseRestrictionConverter useRestrictionConverter;
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final DataAccessRequestDAO dataAccessRequestDAO;
+    private final DataSetDAO dataSetDAO;
+    private final WebTarget matchServiceTarget;
 
-    public static void initInstance(Client client, DataAccessRequestDAO dataAccessRequestDAO,
-                                    ServicesConfiguration config, ConsentDAO consentDAO, ElectionDAO electionDAO, DatasetService datasetService, UseRestrictionConverter useRestrictionConverter) {
-        MatchAPIHolder.setInstance(new DatabaseMatchingServiceAPI(client, dataAccessRequestDAO, config, consentDAO, electionDAO, datasetService, useRestrictionConverter));
-    }
+    private final GenericType<ResponseMatchingObject> rmo = new GenericType<>(){};
 
-    DatabaseMatchingServiceAPI(Client client, DataAccessRequestDAO dataAccessRequestDAO, ServicesConfiguration config, ConsentDAO consentDAO, ElectionDAO electionDAO, DatasetService datasetService, UseRestrictionConverter useRestrictionConverter){
+    @Inject
+    public MatchService(Client client, ServicesConfiguration config, ConsentDAO consentDAO, MatchDAO matchDAO,
+                        ElectionDAO electionDAO, DataAccessRequestDAO dataAccessRequestDAO, DataSetDAO dataSetDAO,
+                        UseRestrictionConverter useRestrictionConverter) {
+        this.matchDAO = matchDAO;
         this.consentDAO = consentDAO;
-        this.dataAccessRequestDAO = dataAccessRequestDAO;
+        this.accessAPI = AbstractDataAccessRequestAPI.getInstance();
```

Comment:
The only need for accessAPI here is for the method:
```
    private void validatePurpose(String purposeId) {
        if (accessAPI.describeDataAccessRequestById(purposeId) == null) {
            throw new IllegalArgumentException("Purpose for the specified id does not exist");
        }
    }
```
However, that can easily be replaced with a call from `dataAccessRequestDAO` which is preferable to using the singleton here.